pennylane/plugins/tests/test_measurements.py

```diff
@@ -0,0 +1,728 @@
+        assert np.allclose(res, expected, atol=tol(dev.analytic))
+
+    def test_hermitian(self, device, tol, skip_if):
+        """Test that a tensor product involving qml.Hermitian works correctly"""
+        n_wires = 3
+        dev = device(n_wires)
+        skip_if(dev, {"tensor_observable": False})
+
+        theta = 0.432
+        phi = 0.123
+        varphi = -0.543
+        A_ = np.array(
+            [
+                [-6, 2 + 1j, -3, -5 + 2j],
+                [2 - 1j, 0, 2 - 1j, -5 + 4j],
+                [-3, 2 + 1j, 0, -4 + 3j],
+                [-5 - 2j, -5 - 4j, -4 - 3j, -6],
+            ]
+        )
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX(theta, wires=[0])
+            qml.RX(phi, wires=[1])
+            qml.RX(phi, wires=[2])
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 2])
+            return qml.expval(qml.PauliZ(wires=0)), qml.expval(qml.Hermitian(A_, wires=[1, 2]))
+
+        res = circuit()
+
+        expected = 0.5 * (
+            -6 * np.cos(theta) * (np.cos(varphi) + 1)
+            - 2 * np.sin(varphi) * (np.cos(theta) + np.sin(phi) - 2 * np.cos(phi))
+            + 3 * np.cos(varphi) * np.sin(phi)
+            + np.sin(phi)
+        )
+        assert np.allclose(res, expected, atol=tol(dev.analytic))
+
+
+@flaky(max_runs=10)
+class TestSample:
+    """Tests for the sample return type."""
+
+    def test_sample_values(self, device, tol):
+        """Tests if the samples returned by sample have
+        the correct values
+        """
+        n_wires = 1
+        dev = device(n_wires)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX(1.5708, wires=[0])
+            return qml.sample(qml.PauliZ(wires=0))
+
+        res = circuit()
+
+        # res should only contain 1 and -1
+        assert np.allclose(res ** 2, 1, atol=tol(False))
+
+    def test_sample_values_hermitian(self, device, tol):
+        """Tests if the samples of a Hermitian observable returned by sample have
+        the correct values
+        """
+        n_wires = 1
+        dev = device(n_wires)
+
+        A_ = np.array([[1, 2j], [-2j, 0]])
+        theta = 0.543
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX(theta, wires=[0])
+            return qml.sample(qml.Hermitian(A_, wires=0))
+
+        res = circuit()
+
+        # res should only contain the eigenvalues of
+        # the hermitian matrix
+        eigvals = np.linalg.eigvalsh(A_)
+        assert np.allclose(sorted(list(set(res))), sorted(eigvals), atol=tol(dev.analytic))
+        # the analytic mean is 2*sin(theta)+0.5*cos(theta)+0.5
+        assert np.allclose(
+            np.mean(res), 2 * np.sin(theta) + 0.5 * np.cos(theta) + 0.5, atol=tol(False)
+        )
+        # the analytic variance is 0.25*(sin(theta)-4*cos(theta))^2
+        assert np.allclose(
+            np.var(res), 0.25 * (np.sin(theta) - 4 * np.cos(theta)) ** 2, atol=tol(False)
+        )
```

Comment:
I've always been unhappy with the way these sample tests are designed; essentially, they are checking that

* The returned samples have the allowed values
* The returned samples have the right statistical properties up to a tolerance 

Do you think this is sufficient? I've never been able to come up with a better end-to-end integration tests for sampling. (If this was a unittest, and we had access to the simulator code, we would be able to simply monkeypatch and check if the deterministic quantities used to generate the samples were correct...)

Reply:
It's our old question. I think we should set a standard for ourselves!

Reply:
I'm not sure there's much else to test :thinking:. You could add more random variables, by the mean and variance seem like the natural things to check